app/helpers/request_helper.py

```diff
@@ -43,12 +43,20 @@ def response_ok(response, service_url=None):
     if response is None:
         logger.error("No response from service")
         return False
+
     elif response.status_code == 200:
         logger.info("Returned from service", request_url=response.url, status=response.status_code, service=service)
         return True
+
+    elif response.status_code == 404:
+        logger.info("NotFound returned from service", request_url=response.url, status=response.status_code,
```

Comment:
I'm not sure this is the correct place for this logic. This method (response_ok) is used for the call to both store and sequence so raising a DocumentNotFound error doesn't quite make sense (though sequence technically wouldn't ever return a 404).

I wonder if it would be better to move the logic into get_doc_from_store and refactor this method to return the actual http error code.